### PR TITLE
Update combat system

### LINKED ISSUE #5: Combat system
- primary weapon attack
 - 1 class ability

### PR DESCRIPTION
**Pull request by JadenBalogh**
Includes core logic for Actors, an extension of Entity which is used for any object that has health and interacts with other Actors. Also has updates for the combat system, demonstrated with an example melee cleave attack and a fireball attack.

Closes #5.

### CODE REVIEW

File: Assets/Scripts/Game/Player.cs [hunk 1]
```diff
@@ -31,7 +32,7 @@ public class Player : Entity
 
     private PositionChangedEvent onPositionChanged = new PositionChangedEvent();
 
-    void Awake()
+    protected override void Awake()
```
Comment (TheArchitect4855) [inline Assets/Scripts/Game/Player.cs:35]: Do you not need to call `base.Awake()` as well?

File: Assets/Scripts/Game/UI/HealthDebugText.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+
+public class HealthDebugText : MonoBehaviour
+{
+    private TextMeshProUGUI text;
```
Comment (TheArchitect4855) [inline Assets/Scripts/Game/UI/HealthDebugText.cs:8]: I'm not saying this is bad, but wouldn't it be easier to use `OnGUI` for debug text?

**Review by TheArchitect4855 — COMMENTED** [review]
Awesome work. I only had a couple of nitpicky things.

I feel like `Promo/melee-attack-demo.mp4` should probably be in its own repo to avoid bloat as it's not something the game itself needs.

I also really like the Action system; it's slick.

File: Assets/Scripts/Game/UI/HealthDebugText.cs (hunk 2 above)
Comment (JadenBalogh) [reply Assets/Scripts/Game/UI/HealthDebugText.cs:8]: Yeah, probably. I didn't even really think about it TBH. For this PR I won't change it, but I'll definitely look into doing it that way in the future for efficiency.

File: Assets/Scripts/Game/Player.cs (hunk 1 above)
Comment (JadenBalogh) [reply Assets/Scripts/Game/Player.cs:35]: Good catch!

**Comment by JadenBalogh** [comment]
> Awesome work. I only had a couple of nitpicky things.
> 
> I feel like `Promo/melee-attack-demo.mp4` should probably be in its own repo to avoid bloat as it's not something the game itself needs.
> 
> I also really like the Action system; it's slick.

Thanks! I agree with the promo stuff. I'll move that into the Google Drive for now.

**Review by TheArchitect4855 — APPROVED** [review]
Pog

**Pull request merged**